bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RuleStatusInfo.java

```diff
@@ -45,20 +75,44 @@ public RuleStatusInfo(RuleStatus status) throws IllegalArgumentException {
      *
      * @throws IllegalArgumentException if status or status detail is null
      */
-    public RuleStatusInfo(RuleStatus status, RuleStatusDetail statusDetail) throws IllegalArgumentException {
-        this(status, statusDetail, null);
+    public static RuleStatusInfo create(RuleStatus status, RuleStatusDetail statusDetail) throws IllegalArgumentException {
+        if(statusDetail == null || statusDetail == statusDetail.NONE) {
+            return create(status);
+        }else {
+            return create(status, statusDetail, null);
+        }    
+    }
+
+    public static RuleStatusInfo create(RuleStatus status, RuleStatusDetail statusDetail, String description) {
+        if(status == null) {
```

Comment:
@triller-telekom Instead of having all those IAEs, wouldn't that be a perfect case to make use of the new Nonnull annotations?